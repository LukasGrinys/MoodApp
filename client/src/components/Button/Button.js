import React from 'react';
import styles from './Button.module.scss';
import { useTheme } from '../../contexts/ThemeContext';
import classNames from 'classnames';

/* 
    Possible colours:
    'primary'
    'white'
*/

const Button = ({
    color = 'primary', 
    handleClick = () => {},
    disabled,
    children
}) => {
    const darkTheme = useTheme();
    
    const getButtonColorClass = (type) => {
        switch (type) {
            case 'primary':
                return styles.btnPrimary;
            case 'white' :
                return styles.btnWhite;
            default :
                return null;
        }
    }
    
    return (
        <button 
            className={classNames(
                styles.button,
                darkTheme && styles.btnDark,
                getButtonColorClass(color)
            )}
            onClick={handleClick}
            disabled={disabled}
            type="button"
        >
            {children}
        </button>
    );
};

export default Button;