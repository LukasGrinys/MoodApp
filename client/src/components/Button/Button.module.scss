@import '../../styles/config/variables';

.button {
    border-radius: 4px;
    font-size: 1.1em;
    font-family: $font-primary;
    width: 150px;
    padding: 5px;
    text-align: center;
    cursor: pointer;
    box-shadow: $shadow-button;
    transition-duration: 0.2s;
    outline: none;
    border: 0;

    &:hover {
        transform: scale(1.02);
    }

    &.btnPrimary {
        color: $color-ivory-white;
        background: $color-primary;
    }

    &.btnWhite {
        background: $color-ivory-white;
        color: $color-primary;
    }

    &.btnDark {
        color: $color-night-dark;
        background: $color-night-lightgrey;
    }

    &:disabled {
        background: $color-disabled-background;
        color: $color-disabled;
        cursor: not-allowed;
    }
}

@media (max-width: 400px) {
    .button {
        width: 120px;
    }
}