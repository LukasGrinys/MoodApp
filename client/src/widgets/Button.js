import React from 'react';
import styles from './button.module.css';
import { useTheme } from './../hoc/ThemeContext';


const ButtonWid = (props) => {
    const darkTheme = useTheme();

    const returnStyle = (darkTheme, isDisabled) => {
        let styleObject = {};
        // Color properties
        if (darkTheme) {
            styleObject.color = "#0d0d0d";
            styleObject.background = "#d2d2d2";
        } else {
            styleObject.color = props.color;
            styleObject.background = props.background;
        }
        // Disabled
        if (isDisabled) {
            styleObject.background = '#cccccc';
            styleObject.color = '#808080';
            styleObject.cursor = 'not-allowed';
        };
        return styleObject;
    }

    return (
        <div  
            style={returnStyle(darkTheme, props.disabled)}
            className={styles.button} 
        >
            {props.children}
        </div>
    );
};

export default ButtonWid;