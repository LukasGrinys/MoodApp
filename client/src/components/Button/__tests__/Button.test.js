import React from 'react';
import Button from '../Button';
import { shallow } from 'enzyme';

const clickMock = jest.fn();

const testProps = {
    handleClick: clickMock,
    disabled: false,
    children: 'Test button'
};

const wrapper = shallow(<Button {...testProps}/>);

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Has button element', () => {
    expect(wrapper.find('button').length).toBe(1);
});

test('Button element has proper props', () => {
    expect(wrapper.find('button').props()).toEqual({
        className: 'button btnPrimary',
        onClick: testProps.handleClick,
        disabled: testProps.disabled,
        type: 'button',
        children: testProps.children
    });
});

test('Button on click calls passed function', () => {
    wrapper.find('button').simulate('click');
    expect(testProps.handleClick).toHaveBeenCalledTimes(1);
});

test('Button changes color if "color" prop is passed', () => {
    const element = shallow(<Button {...testProps} color={"white"}/>);
    expect(element.find('button').prop('className')).toBe('button btnWhite');
})

