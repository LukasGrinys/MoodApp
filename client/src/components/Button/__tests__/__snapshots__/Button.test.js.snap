// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<button
  className="button btnPrimary"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Test button
</button>
`;
